org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/performance/DOMParserPerformance.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4xml.performance;
+
+import java.io.InputStream;
+
+import org.eclipse.lsp4xml.commons.TextDocument;
+import org.eclipse.lsp4xml.dom.DOMDocument;
+import org.eclipse.lsp4xml.dom.DOMParser;
+
+/**
+ * This utility class is used to check memories with use of {@link DOMParser}
+ * with the big file nasa.xml.
+ * 
+ * @author Angelo Zerr
+ *
+ */
+public class DOMParserPerformance {
+
+	public static void main(String[] args) {
+		InputStream in = DOMParserPerformance.class.getResourceAsStream("/xml/nasa.xml");
+		String text = convertStreamToString(in);
+		TextDocument document = new TextDocument(text, "nasa.xml");
+		// Parse every time the big file with DOM parser.
+		while (true) {
+			long start = System.currentTimeMillis();
+			DOMDocument xmlDocument = DOMParser.getInstance().parse(document, null);
+			System.err.println("Parsed 'nasa.xml' with DOMParser in " + (System.currentTimeMillis() - start) + " ms.");
+		}
+	}
+
+	static String convertStreamToString(InputStream is) {

```

Comment:
move to utility class to avoid duplication with XMLScannerPerformance (or create AbstractPerformanceProbe class)

Reply:
@fbricon updated